generators/entity-server/templates/src/main/java/package/repository/EntityRepositoryWithBagRelationshipsImpl.java.ejs

```diff
@@ -0,0 +1,87 @@
+<%#
+ Copyright 2013-2021 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%= packageName %>.repository;
+
+import java.util.List;
+import java.util.Optional;
+<%_ if (fieldsContainUUID) { _%>
+import java.util.UUID;
+<%_ } _%>
+
+import javax.persistence.EntityManager;
+
+import org.hibernate.annotations.QueryHints;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+
+import <%= packageName %>.domain.<%= persistClass %>;
+
```

Comment:
@mshima Can you reference https://vladmihalcea.com/hibernate-multiplebagfetchexception/ in the class comments please?